src/main/java/com/falkordb/impl/resultset/ResultSetImpl.java

```diff
@@ -256,6 +258,16 @@ private Object deserializePoint(Object rawScalarData) {
         return new Point(BuilderFactory.DOUBLE_LIST.build(rawScalarData));
     }
 
+    private List<Float> deserializeVector(Object rawScalarData) {
+        List<byte[]> array = (List<byte[]>) rawScalarData;
+       
+        List<Float> res = new ArrayList<>(array.size());
+        for (byte[] val : array) {
+            res.add(Float.parseFloat(SafeEncoder.encode(val)));
```

Comment:
From a performance stand point is this the best way to go ?

Reply:
yes, since we send the floats as strings and not real binary floats